examples/result_file_dataclasses.py

```diff
@@ -0,0 +1,202 @@
+"""
+Working with result files using Python-native dataclasses.
+"""
+from collections.abc import Iterator
+from operator import attrgetter
+from pathlib import Path
+
+import pendulum
+import polars as pl
+
+from indico import IndicoClient
+from indico.queries import (
+    GetSubmission,
+    SubmissionResult,
+    RetrieveStorageObject,
+    SubmitReview,
+)
+from indico_toolkit import results
+
+
+"""
+Load a result file.
+"""
+client = IndicoClient()
+submission = client.call(GetSubmission(123))  # Or similar.
+result_dict = client.call(RetrieveStorageObject(submission.result_file))
+result = results.load(result_dict)
+
+# If your submission is unreviewed,
+# generate a result file to include review information.
+submission_result = client.call(SubmissionResult(submission, wait=True))
+result_dict = client.call(RetrieveStorageObject(submission_result.result))
+result = results.load(result_dict)
+
+# Or convert the unreviewed submission to a reviewed submission.
+# Due to the lack of review data, only `document.final` is populated.
+result = results.load(result_dict, convert_unreviewed=True)
+
+# `results.load` also takes JSON strings and file paths.
+for result_file in Path("results_folder").glob("*.json"):
+    result = results.load(result_file)
+
+
+"""
+Save document classification and extractions to CSV.
+"""
+
+
+def get_predictions(result: results.Result) -> Iterator[dict[str, object]]:
```

Comment:
I'd add in the assumption here-- i.e. `assumes you have one classification model and one set of extraction results`

Reply:
The updated `results_to_csv.py` script now makes no assumptions on the number of classification or extraction models and instead dumps all of them.